### DISCUSSION TITLE
What is the best tutorial?

### DISCUSSION THREAD

**Discussion by jsbien** [trufanov-nok/scantailor-universal#72] [Q&A]
Hi!
Several days ago I tried scantailor on a 700-pages volume of a dictionary and was totally confused :-( For example, it binarized all the pages (ecept one) although I didn't want to...

**Comment by trufanov-nok** [comment]
There are plenty ScanTailor tutorials but most of them in russian. I think you may start with this video:
https://vimeo.com/12524529
That's a tutorial based on original ScanTailor made by its author Joseph Artsimovich (Tulon on GitHub). And there seems to be a few newer english tutorials on youtube. Let me know if you'll need some additional info.

**Reply by jsbien** [reply]
Thanks. For me it devotes too much attention to the illustrations which I don't care about. With the dictionary (it is in my scope of interest since several years already, cf. e.g. https://szukajwslownikach.uw.edu.pl/slownik-lindego-nowy/, but now I focus on the first edition) I intend to make haste slowly. I noticed several videos on YouTube but I'm afraid they just explain the basic ideas. I will have a closer look at them soon. Which Russian tutorial is most detailed?

**Reply by trufanov-nok** [reply]
Well, It would be great if you share a direct link to djvu document as example. If I recall right, it's quite a specific: it tries to keep original book look and feel as much as possible. I mean background etc. ST is designed for other purposes. It produces either color pages (color mode) which useful for book covers or binarized images (b/w mode) or binarized images with color illustrations (mixed mode). Where "binarized image" is a black text on **white** background (can be later change to other solid color, like yellowish). So if you're not interested in illustrations:
> For example, it binarized all the pages (ecept one) although I didn't want to...

That's exactly what you should expected to get - binarized pages.  

Apart from that, current ST and its forks has little about djvu dictionaries. Their results must be encoded with one of djvu encoders which will explicitly (minidjvu/cjb2) or internally (Lizardtech's encoder) work with binarized images. So in a nutshell ST is a semi-manual segmentator that splits illustrations and background from text layer. And that's user's and djvu encoder's task how to combine them back.   

For ex in your case, most probably, you'll need to get those binarized images, encode them into djvu, then with help of DjVuLibre's tools put original (downscaled? I wouldn't recommend to downscale illustrations) image as a background of each page and may be to perform the trick I saw in [`didjvu`](https://github.com/trufanov-nok/djvudict/issues/1#issuecomment-627463707): using binarized image as a mask to cut out character's glyphs from original image, downscale them and use that low-sampled image to "colorize" text layer.  To keep look and feel max close to original book. It's a question what image they use or is better to use:  the one in place of original character or the one that was chosen as a pattern for djbz dictionary...

So I would say you may need ST as a segmentator that can not only separate text layer from others but also allow to separate real illustrations from a background. And you should produce a mixed-mode or bw-mode images: on white background. Then you shall examine Export functionality of ST Universal to split those images into bw text (on white background)  and illustrations (non-bw on white background). Then you may use these 2 layers as masks to decompose source scans into background, real illustrations and text - all in original colors. So we have 5 layers now (in fact - 4). Than it can be combined back into djvu with fancy scripts. Perhaps slightly changed DjVu Imager (that i [recreated](https://github.com/trufanov-nok/DjVu_Imager-Qt) with Qt) may be used to automate that. In theory STU may do this itself in future (a distant) after integration with djvu encoder.  And all of that has little to djbz encoding. For djbz encoding you need only binarized text and encoder.   
That's all I can say bcs I saw the djvu document sample that you most probably expects to have at the end, so direct link to the sample document would be more informative to others.

**Reply by jsbien** [reply]
Thanks for your long comments. They will be very useful but rather at the later stage of my work.

As for the direct links to the dictionary, there are three scans which I can work on. One is in KPBC (https://kpbc.umk.pl/publication/8173 ), but the quality is rather low, in particular dewarping is definitely necessary. The second is a 600 dpi scan at Crispa (https://crispa.uw.edu.pl/object/files/167261/) - one volume missing, one in PDF, the rest in DjVu. Hence I start my experiments with the third scans from Polona (e.g. [Tom 1 Część 1 (1807)](https://polona.pl/item/2632461)) originally served with IIIF. With the help of [dezoomify-rs](https://github.com/lovasoa/dezoomify-rs) I downloaded what seems to be the master JPG files. I consider uploading the files if there will be some demand for it (Polona allows to redistribute public domain works).

I made a quick test of ScanTailor because I new it is "politically correct" to use it (i.e. practically all recommend it), but I want to move slowly and my next step is just deskewing. I'm aware of unpaper and some other tools found with Google, but  I was curious how it works in ScanTailor.

BTW, you mentioned didjvu. I will start another thread about it.

**Comment by Piolie** [comment]
There is also the [`A Linux Guide to Book Scanning`](https://natecraun.net/articles/linux-guide-to-book-scanning.html), by [Nate Craun](https://github.com/ncraun). It's in English, though very brief.

**Reply by jsbien** [reply]
Thanks, but I'm looking for something which covers the subtleties of the program, not just the general ideas.

**Comment by Noitaenola** [comment]
I wrote a long guide in Spanish. Have a look at: https://github.com/Noitaenola/ScanManual

**Reply by jsbien** [reply]
Thanks. Don't know Spanish but will have a look.
